corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -171,7 +171,8 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         var self = this;
 
         if (!json.type) {
-            Container.groupQuestions(json);
+            Container.groupElements(json, constants.QUESTION_TYPE);
```

Comment:
Does this first group questions and then goes through again and groups the groups? 

What if a row could fit 2 groups and a question? I think all children needs to be grouped sequentially but i'm not sure. For example: if the form is like this
Question (width 4)
Question (width 4)
Group (width 4)
They should all fit on the same row. But if we group Question first, then it'll put the two questions on one row (it won't fit the group since it's still width =12).

Reply:
> Does this first group questions and then goes through again and groups the groups?

Yes

> What if a row could fit 2 groups and a question? 

That's a good question. I didn't think about this scenario and was more inclined towards the case shared in the ticket(two or more groups in a row). Right now sequential questions are grouped under 'grouped-question' and I have added logic for the groups to be under 'grouped-groups'. When there are different types of elements in the same row, there need to be a common grouping type in order to keep them in that row using `<div class='row'>`. How could that be achieved, any suggestion?